prelude.dx

```diff
@@ -801,24 +801,47 @@ String : Type = List Char
 
 CharPtr : Type = %CharPtr
 
-interface Show a:Type where
-  show : a -> String
-
 -- TODO. This is ASCII code point. It really should be Int32 for Unicode codepoint 
 def codepoint (c:Char) : Int8 = %codePoint c
 
 @instance charEq  : Eq Char  = MkEq \x y. codepoint x == codepoint y
 @instance charOrd : Ord Char = (MkOrd charEq (\x y. codepoint x > codepoint y)
                                              (\x y. codepoint x < codepoint y))
 
+interface Show a:Type where
+  show : a -> String
 
-def showFloat' (x:Float) : String =
-  (n, ptr) = %ffi showFloat (Int & CharPtr) x
+def showInt32' (x:Int32) : String =
+  (n, ptr) = %ffi showInt32 (Int & CharPtr) x
   AsList n $ for i:(Fin n).
     MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
 
-instance showFloat : Show Float where
-  show = showFloat'
+def showInt64' (x:Int64) : String =
+  (n, ptr) = %ffi showInt64 (Int & CharPtr) x
+  AsList n $ for i:(Fin n).
+    MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+def showFloat32' (x:Float32) : String =
+  (n, ptr) = %ffi showFloat32 (Int & CharPtr) x
+  AsList n $ for i:(Fin n).
+    MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+def showFloat64' (x:Float64) : String =
+  (n, ptr) = %ffi showFloat64 (Int & CharPtr) x
+  AsList n $ for i:(Fin n).
+    MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+instance showInt32 : Show Int32 where
+  show = showInt32'
+
+instance showInt64 : Show Int64 where
+  show = showInt64'
+
+instance showFloat32 : Show Float32 where
+  show = showFloat32'
+
+instance showFloat64 : Show Float64 where
+  show = showFloat64'
```

Comment:
Maybe consider inlining the definitions into those instances? There's no reason to pollute the namespace, especially that we don't have any way to restrict what prelude exports.

Reply:
Good point. Done.